### PR TITLE
Replace 'exact' lookup_type with 'icontains' in Groups API.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by johngian** [comment]
Since ``groups`` and ``skills`` models share some similarities I suggest we do the same to ``skills`` as well.

**Comment by johngian** [comment]
Since we are changing our public api we should update the docs as well.

**Comment by akatsoulas** [comment]
@johngian  pr updated. Thanks!

**Comment by johngian** [comment]
By changing the default lookup type for these fields we are braking the functionality for our current consumers.

I suggest we:
* Leave the default `lookup_type` as it is
* Add an extra `lookup_type` that introduces the new functionality.

**Comment by akatsoulas** [comment]
@johngian pr updated

**Comment by johngian** [comment]
Instead of adding extra fields its better to extra lookup_types. This gives us flexibility to build more on top.

* https://django-filter.readthedocs.org/en/latest/usage.html#the-filter
 * Multiple lookup types

**Comment by akatsoulas** [comment]
@johngian pr updated

File: mozillians/groups/api/v2.py [hunk 1]
```diff
@@ -76,6 +76,14 @@ class Meta:
         model = Group
         fields = ('name', 'functional_area', 'curators', 'curator',
                   'members_can_leave', 'accepting_new_members',)
+        fields = {
+            'name': ['exact', 'icontains'],
+            'functional_area': ['exact'],
```
Comment (johngian) [inline mozillians/groups/api/v2.py]: Isn't exact the default anyway?

File: docs/api/apiv2/api-groups.rst [hunk 2]
```diff
@@ -22,7 +22,9 @@ Parameters
         *Required* **string** - The application's API key
 
     ``name``
-        *Optional* **string** - Return results matching given name
+        *Optional* **string** - Return results matching the given name
+
+        *Available filters* **icontains** - Return results containing the given name
```
Comment (johngian) [inline docs/api/apiv2/api-groups.rst]: Please explain how filters are applied in the query.

File: mozillians/groups/api/v2.py [hunk 3]
```diff
@@ -76,6 +76,14 @@ class Meta:
         model = Group
         fields = ('name', 'functional_area', 'curators', 'curator',
```
Comment (johngian) [inline mozillians/groups/api/v2.py]: This is redefined two lines after.

**Comment by johngian** [comment]
r+

**Pull request merged**